File: initial/src/main/java/hello/MvcConfig.java
package hello;

import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.ViewControllerRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Deprecated
@Configuration
public class MvcConfig implements WebMvcConfigurer {

	@Deprecated
    /**
     * @deprecated ViewContorollerのRequestMappingに変更。
     * @see hello.app.controller.ViewController 
     */
    public void addViewControllers(ViewControllerRegistry registry) {
//        registry.addViewController("/home").setViewName("home");
//        registry.addViewController("/").setViewName("home");
//        registry.addViewController("/hello").setViewName("hello");
//        registry.addViewController("/login").setViewName("login");
    }

}

File: complete-kotlin/src/main/kotlin/com/example/securingweb/MvcConfig.kt
package com.example.securingweb

import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.ViewControllerRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class MvcConfig : WebMvcConfigurer {

    override fun addViewControllers(registry: ViewControllerRegistry) = with(registry) {
        addViewController("/home").setViewName("home")
        addViewController("/").setViewName("home")
        addViewController("/hello").setViewName("hello")
        addViewController("/login").setViewName("login")
    }
}
